File: zero_jobs_income.py
#!/usr/bin/env python3
"""
Script để đặt thu nhập Jobs về 0 (chỉ giữ lại points và experience)
Sử dụng: python3 zero_jobs_income.py
"""

import yaml
import os
import glob

def zero_job_income(file_path):
    """Đặt tất cả income = 0 trong file job"""
    with open(file_path, 'r', encoding='utf-8') as f:
        data = yaml.safe_load(f)
    
    if not data:
        return
    
    modified = False
    job_name = list(data.keys())[0]  # Tên job (Miner, Builder, etc.)
    job_data = data[job_name]
    
    # Đặt công thức income = 0
    if 'income-progression-equation' in job_data:
        job_data['income-progression-equation'] = '0'
        modified = True
    
    # Duyệt qua các action types (Break, Place, Kill, etc.)
    for action_type in ['Break', 'Place', 'Kill', 'Craft', 'Smelt', 'Repair', 'TNTBreak', 'Tame', 'Breed', 'Milk', 'Shear', 'Fish', 'Enchant', 'custom-kill']:
        if action_type in job_data:
            for item_name, item_data in job_data[action_type].items():
                if isinstance(item_data, dict) and 'income' in item_data:
                    item_data['income'] = 0
                    modified = True
    
    if modified:
        with open(file_path, 'w', encoding='utf-8') as f:
            yaml.dump(data, f, default_flow_style=False, allow_unicode=True)
        print(f"✅ Zeroed income: {os.path.basename(file_path)}")

def main():
    # Tìm tất cả file job
    jobs_dir = "/Users/dev/code/mc/plugin-mc/Jobs/jobs"
    job_files = glob.glob(os.path.join(jobs_dir, "*.yml"))
    
    # Loại bỏ file example và none
    job_files = [f for f in job_files if not any(x in os.path.basename(f).lower() for x in ['example', 'none'])]
    
    print(f"🚫 Setting all Jobs income to 0...")
    print(f"📁 Found {len(job_files)} job files")
    
    for job_file in job_files:
        zero_job_income(job_file)
    
    print(f"✅ Done! All Jobs now give 0 income (only points & experience)")
    print("💡 Players will need to use shop or other methods to earn money")

if __name__ == "__main__":
    main()

File: zero_jobs_income.js
#!/usr/bin/env node
/**
 * Script để đặt thu nhập Jobs về 0 (chỉ giữ lại points và experience)
 * Sử dụng: node zero_jobs_income.js
 * hoặc: npm run zero-income
 */

const fs = require('fs-extra');
const path = require('path');
const yaml = require('js-yaml');
const { glob } = require('glob');

/**
 * Đặt tất cả income = 0 trong file job
 * @param {string} filePath - Đường dẫn file job
 */
async function zeroJobIncome(filePath) {
    try {
        // Đọc file YAML
        const fileContent = await fs.readFile(filePath, 'utf8');
        const data = yaml.load(fileContent);
        
        if (!data) {
            console.log(`⚠️  File rỗng: ${path.basename(filePath)}`);
            return;
        }
        
        let modified = false;
        const jobName = Object.keys(data)[0]; // Tên job (Miner, Builder, etc.)
        const jobData = data[jobName];
        
        // Đặt công thức income = 0
        if (jobData['income-progression-equation']) {
            jobData['income-progression-equation'] = '0';
            modified = true;
        }
        
        // Các action types có thể có trong Jobs
        const actionTypes = [
            'Break', 'Place', 'Kill', 'Craft', 'Smelt', 'Repair', 
            'TNTBreak', 'Tame', 'Breed', 'Milk', 'Shear', 'Fish', 
            'Enchant', 'custom-kill', 'Explore', 'custom-explore', 
            'Brew', 'Brush', 'Strip', 'Dye', 'Collect'
        ];
        
        // Duyệt qua các action types
        for (const actionType of actionTypes) {
            if (jobData[actionType]) {
                for (const [itemName, itemData] of Object.entries(jobData[actionType])) {
                    if (typeof itemData === 'object' && itemData.income !== undefined) {
                        const oldIncome = itemData.income;
                        itemData.income = 0;
                        if (oldIncome !== 0) {
                            console.log(`    ${actionType}.${itemName}: ${oldIncome} → 0`);
                            modified = true;
                        }
                    }
                }
            }
        }
        
        if (modified) {
            // Ghi lại file với format đẹp
            const yamlStr = yaml.dump(data, {
                defaultFlowStyle: false,
                lineWidth: -1,
                noRefs: true,
                sortKeys: false
            });
            
            await fs.writeFile(filePath, yamlStr, 'utf8');
            console.log(`✅ Zeroed income: ${path.basename(filePath)}`);
        } else {
            console.log(`ℹ️  No income found: ${path.basename(filePath)}`);
        }
        
    } catch (error) {
        console.error(`❌ Error processing ${path.basename(filePath)}:`, error.message);
    }
}

/**
 * Main function
 */
async function main() {
    try {
        console.log('🚫 Setting all Jobs income to 0...');
        
        // Tìm tất cả file job
        const jobsDir = path.join(__dirname, 'Jobs', 'jobs');
        const pattern = path.join(jobsDir, '*.yml');
        const jobFiles = await glob(pattern);
        
        // Loại bỏ file example và none
        const validJobFiles = jobFiles.filter(file => {
            const basename = path.basename(file).toLowerCase();
            return !basename.includes('example') && !basename.includes('none');
        });
        
        console.log(`📁 Found ${validJobFiles.length} job files`);
        
        if (validJobFiles.length === 0) {
            console.log('❌ No valid job files found!');
            console.log(`📂 Looking in: ${jobsDir}`);
            return;
        }
        
        // Xử lý từng file
        for (const jobFile of validJobFiles) {
            await zeroJobIncome(jobFile);
        }
        
        console.log('\n✅ Done! All Jobs now give 0 income (only points & experience)');
        console.log('💡 Players will need to use shop or other methods to earn money');
        console.log('🔄 Remember to restart your server to apply changes!');
        
    } catch (error) {
        console.error('❌ Fatal error:', error);
        process.exit(1);
    }
}

// Chạy script nếu được gọi trực tiếp
if (require.main === module) {
    main();
}

module.exports = { zeroJobIncome };
